camel-plantuml-jar/src/main/java/io/github/ncasaux/camelplantuml/generator/RoutesDiagramGenerator.java

```diff
@@ -47,6 +48,11 @@ public static String generateUmlString(HashMap<String, RouteInfo> routesInfo) th
                 }
             }
 
+            if (parameters.uriFilterPattern().matcher(routeEndpointBaseUri).matches()) {
```

Comment:
If we want to filter based on the route Id, I think we should not use the `routeEndpointBaseUri` but the `routeId` instead.